lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -280,6 +282,27 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
     |> propagate_null_trimming
   end
 
+  defp maybe_emit_telemetry_events(res) do
+    end_time_mono = System.monotonic_time()
+    resolution_path = Absinthe.Resolution.path(res)
+
+    with {start_time, start_time_mono} <- res.acc[{:telemetry, resolution_path}] do
+      duration = System.convert_time_unit(end_time_mono - start_time_mono, :native, :millisecond)
+
+      metadata = %{
+        type: :absinthe_resolver_call,
+        start_time: start_time,
+        schema: res.schema,
+        path: resolution_path,
+        field_name: res.definition.name,
+        field_type: Absinthe.Type.name(res.definition.schema_node.type, res.schema),
+        parent_type: res.parent_type.name
+      }
+
+      :telemetry.execute([:absinthe, :resolver, :call], duration, metadata)
+    end
+  end
```

Comment:
I'm really not sure that I'm comfortable making telemetry non optional this way. This function runs for absolutely every field in the entire schema. There's nothing stopping someone from adding telemetry middleware to every field, that's a one liner with `def middleware` or with 1.5 decorators. Baking it into resolution makes it impossible to remove without forking the resolution phase.

Reply:
Do you want it configurable in general, or is this concern simply about preventing it from running for every single field resolution? (which I also agree is critical to avoid)

Reply:
Would using a macro to make that choice configurable at compile time like [` compile_time_purge_matching`](https://hexdocs.pm/logger/Logger.html#module-application-configuration) mean you'd then have [two problems](http://regex.info/blog/2006-09-15/247)?